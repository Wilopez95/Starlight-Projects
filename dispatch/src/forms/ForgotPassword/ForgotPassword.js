import PropTypes from 'prop-types';
import { PureComponent } from 'react';
import { Field, Formik } from 'formik';
import FormError from '../elements/form-error';

class ForgotPassword extends PureComponent {
  static propTypes = {
    handleSubmitForm: PropTypes.func.isRequired,
    onSuccess: PropTypes.func.isRequired,
  };

  constructor(props) {
    super(props);

    this.state = {
      error: false,
      disabled: false,
    };
    this.onValidSubmit = this.onValidSubmit.bind(this);
  }

  async onValidSubmit({ email }) {
    const { handleSubmitForm, onSuccess } = this.props;

    this.disabledSubmit();
    this.setState({ error: false });

    try {
      await handleSubmitForm(email);
      this.enableSubmit();
      onSuccess();
      // history.push('/login')
    } catch (error) {
      this.enableSubmit();
      this.setState({ error });
    }
  }

  enableSubmit = () => {
    this.setState({ disabled: false });
  };

  disabledSubmit = () => {
    this.setState({ disabled: true });
  };

  validateEmail(email) {
    const re =
      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
    return re.test(String(email).toLowerCase());
  }

  render() {
    return (
      <Formik
        enableReinitialize
        initialValues={{
          email: '',
        }}
        onSubmit={(values) => {
          this.onValidSubmit(values);
        }}
      >
        {({ values, handleSubmit }) => (
          <form className="form__login" style={{ backgroundColor: 'white', textArea: 'white' }}>
            {this.state.error ? <FormError error={this.state.error} /> : null}
            <div
              className="input-container"
              style={{
                display: 'flex',
                width: '100%',
                marginBottom: '15px',
              }}
            >
              <i
                className="fa fa-envelope icon"
                style={{
                  padding: '10px',
                  background: '#e87900',
                  color: 'white',
                  minWidth: '50px',
                  textAlign: 'center',
                }}
              />
              <Field
                name="email"
                type="email"
                className="input-field"
                style={{ width: '100%', padding: '10px', outline: 'none' }}
              />
            </div>
            <button
              type="submit"
              onClick={handleSubmit}
              className="btn btn__success btn__lg"
              disabled={this.state.disabled || !this.validateEmail(values.email)}
              style={{
                width: '100%',
                color: 'white',
                textAlign: 'center',
                fontSize: '1.2em',
                backgroundColor: '#e87900',
                borderRadius: '0px',
                marginBottom: '10px',
              }}
            >
              Send recovery email
            </button>
          </form>
        )}
      </Formik>
    );
  }
}

export default ForgotPassword;
