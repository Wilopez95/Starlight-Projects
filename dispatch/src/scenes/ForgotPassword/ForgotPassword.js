/* eslint-disable no-irregular-whitespace */
import { Component } from 'react';
import { Link } from 'react-router-dom';
import styled from 'styled-components';
import PropTypes from 'prop-types';
import Footer from '@root/components/Footer';
import { restore } from '@root/helpers/auth';
import ForgotPasswordForm from '../../forms/ForgotPassword';
import styles from './style.module.css';

const LoginPage = styled.div`
  width: 100%;
  position: relative;
  margin: 0;
  height: auto !important;
  align-items: center;
`;

const LoginWrapper = styled.div`
  width: 510px;
  padding-top: 75px;
  margin: 100px auto 0px auto;
  text-align: center;
  background-color: #fff;
  border-radius: 4px;
  height: auto !important;
  box-shadow: 0 2px 8px 0 rgba(33, 43, 54, 0.1);
  form {
    margin-bottom: 14px;
  }
`;

const LogoArea = styled.div`
  text-align: center;
  margin-bottom: 10px;
`;

const LogoImg = styled.img`
  display: inline-block;
  vertical-align: top;
  max-width: 200px;
`;

const LoginFormFooter = styled.div`
  width: 100%;
  /* border-radius: 4px; */
  padding: 0.5em;
  border: solid 1px #dfe3e8;
  background-color: #f9fafb;
`;

const LoginText = styled.h1`
  color: #e87900;
  padding-top: 20px;
  font-size: 20px;
  text-align: center;
  font-weight: 700;
  paddingbottom: 20px;
`;

// type Props = {
//   history: History,
// };

class ForgotPassword extends Component {
  static displayName = 'ForgotPassword';

  static propTypes = {
    history: PropTypes.object,
  };

  constructor(props) {
    super(props);

    this.state = { submitted: false };
  }

  onNavigateHome = () => {
    setTimeout(() => {
      this.props.history.push('/login');
    }, 5000);
  };

  showSuccessMessage = () => {
    this.setState((prevState) => ({
      submitted: !prevState.submitted,
    }));
    // after successful email send redirect after user has time to read validation back to login page)
    this.onNavigateHome();
  };

  render() {
    return (
      <LoginPage>
        <LoginWrapper>
          <LogoArea>
            <LogoImg
              src="https://cdn.starlightpro.com/starlight-logo-plain.png"
              alt="starlightpro logo"
            />
          </LogoArea>
          <LoginText className="login-text">Reset Password</LoginText>
          {this.state.submitted ? (
            <div data-name="success" className={styles.restorePasswordSuccess}>
              <p>
                <strong>Success!</strong> We just sent you an email with a link to setup your new
                password.
                <br />
                We are redirecting you back to login shortly. If you are not redirected please{' '}
                <Link to="/login">click here</Link>
              </p>
            </div>
          ) : (
            <div className={styles.restorePasswordForm}>
              <ForgotPasswordForm
                handleSubmitForm={restore}
                onSuccess={this.showSuccessMessage}
                onClick={this.onNavigateHome}
              />
            </div>
          )}
          <LoginFormFooter>
            <Link to="/login">Back to Login Screen</Link>
          </LoginFormFooter>
        </LoginWrapper>
        <Footer />
      </LoginPage>
    );
  }
}
export default ForgotPassword;
