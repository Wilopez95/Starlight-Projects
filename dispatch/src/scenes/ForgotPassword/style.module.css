.formContainer {
  width: 510px;
  padding-top: 100px;
  margin: 0px auto 0px auto;
  text-align: center;
}

.forgotPwHeader {
  color: #1fa67b;
  padding-top: 20px;
  font-size: 20px;
  text-align: center;
  font-weight: bold;
  padding-bottom: 20px;
}
.loginLogo {
  text-align: center;
  margin-bottom: 20px;
}
.forgotLogoImg {
  display: inline-block;
  vertical-align: top;
  max-width: 200px;
}
.restorePasswordSuccess {
  text-align: center;

  h2 {
    font-size: 12px;
    margin-bottom: 10px;
    margin-left: 10px;
    margin-right: 10px;
    margin-top: 10px;
  }

  p {
    font-size: 1em;
    margin-bottom: 14px;
    margin-left: 10px;
    margin-right: 10px;
    margin-top: 3px;
  }
  strong {
    font-weight: 700;
  }
}

.restorePasswordForm {
  text-align: center;

  form {
    margin-bottom: 14px;

    label {
      text-align: left;
    }
  }

  > p {
    font-size: 14px;
  }
}
