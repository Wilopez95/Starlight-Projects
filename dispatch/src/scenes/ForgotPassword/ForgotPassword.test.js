import { shallow } from 'enzyme';
import ForgotPassword from './ForgotPassword';

describe('<ForgotPassword />', () => {
  it('renders without crashing', () => {
    const wrapper = shallow(<ForgotPassword />);

    expect(wrapper).toMatchSnapshot();
  });
  it('its state is unsubmitted by default', () => {
    const wrapper = shallow(<ForgotPassword />);

    expect(wrapper.state().submitted).toEqual(false);
  });
});
