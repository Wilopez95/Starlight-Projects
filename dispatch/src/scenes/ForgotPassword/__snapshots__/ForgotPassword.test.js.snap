// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ForgotPassword /> renders without crashing 1`] = `ShallowWrapper {}`;
